proof/invariant-abstract/Invariants_AI.thy

```diff
@@ -3156,10 +3209,94 @@ lemma vms_ioc_update[iff]:
   "valid_machine_state (is_original_cap_update f s::'z::state_ext state) = valid_machine_state s"
   by (simp add: valid_machine_state_def)+
 
+lemma valid_ioc_sched_act_update[simp]:
+  "valid_ioc (scheduler_action_update f s) = valid_ioc s"
+  by (simp add: valid_ioc_def)
+
+lemma valid_ioc_ready_queues_update[simp]:
+  "valid_ioc (ready_queues_update f s) = valid_ioc s"
+  by (simp add: valid_ioc_def)
+
+lemma valid_ioc_domain_time_update[simp]:
+  "valid_ioc (domain_time_update f s) = valid_ioc s"
+  by (simp add: valid_ioc_def)
+
+lemma cur_tcb_cdt_update [simp]:
+  "cur_tcb (cdt_update f s) = cur_tcb s"
+  by (simp add: cur_tcb_def)
+
+lemma cur_tcb_more_update[iff]:
```

Comment:
do you know why this one is `[iff]` when all the other ones are simp?
It also looks super repetitive, they all have nearly the same proof... it almost has a crunch-like flavour, but I'm not sure we have a mechanism that can do that. Maybe scrunching the proofs into one for `cur_tcb` and another for `valid_ioc`?

Reply:
Not sure, presumably historical reasons of whoever wrote them preferring one attribute over the other. I guess `simp` would be the preference now if we wanted to make them consistent?

Reply:
I'm not sure what the difference is between `[iff]` and `[simp]` to be honest. If it's the same, simp will be more consistent, but ok to leave it.

Reply:
`[iff]` also makes `[intro!]` and `[elim!]` rules out of the equation, so e.g. `clarify` would act as well on that constant. Could be that this specific one has one such instance, or it was just done by another person than the rest.